server/src/main/java/org/opensearch/identity/UserIdentity.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.util.List;
+
+/**
+ * An authenticated user in the system.
+ */
+public class UserIdentity implements Identity {
+
+    /** No identity information or identity information is unavaliable */
+    public static final AnonymousUser ANONYMOUS = new AnonymousUser();
+
+    /** A persistant identifier that is unqiue to the user */
+    private String id;
+
+    public UserIdentity(final String id) {
+        this.id = id;
+    }
+
+    /** A durable and reusable identifier of this user */
+    public String getId() {
+        return this.id;
+    }
+
+    /** No identity information or identity information is unavaliable */
+    public static final class AnonymousUser extends UserIdentity {
+        static final String ID = "Anonymous";
+        protected AnonymousUser() {
```

Comment:
The `protected` modifier doesn't match `final` on the class. This class must not be allowed to be inherited.